libs/header/p2p/session.go

```diff
@@ -172,6 +198,10 @@ func (s *session[H]) doRequest(
 
 // processResponse converts HeaderResponse to Header.
 func (s *session[H]) processResponse(responses []*p2p_pb.HeaderResponse) ([]H, error) {
+	if len(responses) == 0 {
+		return nil, errEmptyResponse
+	}
+
```

Comment:
Because of this, ErrNotFound will never be the case, and peers who told us they don't have anything, will have their score decreased. 

It's actually might be a good thing to do, but then we should unify errEmptyResponse to be ErrNotFound case. Instead of having them to be separate

Reply:
You are not right. We will receive a single message with ErrNotFound statusCode. This case helps to prevent case when the malicious server will close the connection without giving us any header

Reply:
Got it. We still may want to decrease the score of a peer, because it does have nothing and we don't really want to request it again.

Reply:
Don't have to happen in this PR, but can be parked on the issue